roles/core/repositories_client/tasks/centos_8.yml

```diff
@@ -30,14 +32,27 @@
   with_items: "{{ repositories | union(external_repositories|default([])) }}"
   when: ( item != 'os' ) and ( item.name is not defined or item.name != 'os' )
 
-- name: "Setting repositories os ({{ item.1 }})"
+- name: "Setting OS repositories"
+  # CentOS configure BaseOS in /etc/yum.repos.d/CentOS-Base.repo and AppStream
+  # in /etc/yum.repos.d/Centos-AppStream.repo. These are the operating system
+  # repositories, thus installed with the special repository 'os' in
+  # BlueBanquise.
   yum_repository:
-    name: "{{ item.1 }}"
-    description: "{{ item.1 }} gen by Ansible"
-    baseurl: "{{ item.0.baseurl | default(baseurl_prefix + 'os/' + item.1) }}"
-    enabled: "{{ item.0.enabled | default(1) }}"
-    gpgcheck: "{{ item.0.gpgcheck | default(0) }}"
-    gpgkey: "{{ item.0.gpgkey | default(omit) }}"
-    proxy: "{{ item.0.proxy | default(omit) }}"
-  loop: "{{ repositories | product(['AppStream', 'BaseOS']) | list }}"
-  when: ( item.0 == 'os' ) or ( item.0.name is defined and item.0.name == 'os' )
+    name: "{{ item.repoid }}"
+    description: "{{ item.repoid }} gen by Ansible"
+    baseurl: "{{ item.baseurl | default(baseurl_prefix + 'os/' + item.repoid) }}"
+    file: "{{ item.file }}"
+    enabled: "{{ item.enabled | default(1) }}"
+    gpgcheck: "{{ item.gpgcheck | default(0) }}"
+    gpgkey: "{{ item.gpgkey | default(omit) }}"
+    proxy: "{{ item.proxy | default(omit) }}"
+  loop:
+  # Combine BaseOS and AppStream with 'os' parameters if any. Otherwise, this
+  # will configure BaseOS and AppStream with default values.
+    - "{{ repositories|selectattr('name', 'defined')
```

Comment:
I can deactivate BaseOS and AppStream from CentOS-Base.repo and CentOS-AppStream.repo and let the task write BaseOS.repo and AppStream.repo as previously. This is what I did first in #88 (attempt to solve #67). Please note this will add the following messages for each dnf command:
```
Repository AppStream is listed more than once in the configuration
Repository BaseOS is listed more than once in the configuration
```
This is to remove these messages that I changed the behaviour to overwrite existing files for BaseOS and AppStream.
Please advise.

Reply:
Got it. Thanks for rising that.
Let me grab few data and think about it.

Reply:
Only possibility found is to slightly rename local repositories in files (ex: BaseOS_local/AppStream_local) then disabling BaseOS and AppStream like with Centos 7.

dnf would not display the warning.

Simple task, allow keeping reference CentOS files.
But need to be explained in task and readme.

Reply:
I'm not sure disabling CentOS 8 repos like I do for CentOS 7 will answer your concern.
If my understanding is correct, you find the loop too complex.
I can disable BaseOS and AppStream in the first task, then rename BlueBanquise's managed 'os' repositories to BaseOS_local and AppStream_local, which it would result in the following loop:

```
  loop:
  # Combine BaseOS and AppStream with 'os' parameters if any. Otherwise, this
  # will configure BaseOS and AppStream with default values.
    - "{{ repositories|selectattr('name', 'defined')
          |selectattr('name', 'match', 'os')|list
          |combine('repoid':'BaseOS_local'}) }}"
    - "{{ repositories|selectattr('name', 'defined')
          |selectattr('name', 'match', 'os')|list
          |combine('repoid':'AppStream_local'}) }}"
```

That's the same.

The issue here is the magic 'os' repository we have to split in two repositories BaseOS and AppStream because of RHEL 8. The simplest solution would be to remove 'os' from repositories and add both BaseOS and AppStream instead:
```
repositories:
  - bluebanquise
  - AppStream
  - BaseOS
```
It's simple, but has obvious drawbacks: we need a dedicated repositories definition for EL 8 like distribution.

Reply:
Since ec9e900c, we can define repositories with simple items, or hashes:

```
  repositories:
    - bluebanquise
    - { name: 'os', gpgcheck: 1 }  # Flow style
```

So, to handle this special EL 8 case ('os' split), we first need to find if the 'os' repository is configured as a simple item or a hash:
` when: ( 'os' in repositories ) or ( item.name is defined and item.name == 'os' )`

Then in the loop, we need to combine defined parameters (if applicable) to BaseOS and AppStream. For this, we need to:

1. filter repositories and select all entries where attribute 'name' is defined
2. filter the result to select the entry where the name is 'os'
3. Change it to a list
4. Combine the list with repoid.

Which change:

```
repositories:
 - { name: 'os', gpgcheck: 1 }
```
To:

```
repositories:
 - { name: 'os', gpgcheck: 1, repoid: 'BaseOS' }
 - { name: 'os', gpgcheck: 1, repoid: 'AppStream' }
```

Reply:
By the way, ec9e900 is mandatory for usage in my company, we need to be able to enable gpgcheck per repository. I need to extend it to support the dnf/yum exclude parameter as well in a near future (waiting for current PR to be approved/rejected to avoid merge conflict and rebase).